account_invoice_structured_communication/models/account_invoice.py

```diff
@@ -0,0 +1,9 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields, api
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    provider_reference = fields.Char(string="Provider reference invoice",
```

Comment:
How do you get the reference? Is it a normal field on the forms?

In that case, remove the `store` parameter which is used to store `computed` fields.